src/main/java/com/powsybl/openloadflow/network/impl/PropagatedContingency.java

```diff
@@ -450,10 +460,10 @@ public Optional<LfContingency> toLfContingency(LfNetwork network) {
                         boolean otherSideConnected;
                         if (branch.getBus1() == busToLost) {
                             otherSideBus = branch.getBus2();
-                            otherSideConnected = branch.isConnectedSide2();
+                            otherSideConnected = branch.isConnectedSide2() && isConnectedAfterContingencySide2(branchesToOpen, branch);
```

Comment:
Why don't we update the network g=following the contingency?

Reply:
This is done in the `apply` method of  `LfContingency`.

Reply:
Yes, exactly. Note that this method `apply` is not called always in some DC sensitivity analyses, that is why we have a lot of treatments in the `PropagatedContingency` class.